lang/elaborator/src/result.rs

```diff
@@ -35,6 +35,8 @@ pub enum TypeError {
         lhs_span: Option<SourceSpan>,
         #[label("Source of (2)")]
         rhs_span: Option<SourceSpan>,
+        #[label("While elaborating")]
+        reason_span: Option<SourceSpan>,
```

Comment:
I'm wondering if there's a better term than `reason` to describe these fields. As far as I can see, this is always the span of the syntax tree node that is currently being elaborated. 
I'm not sure I have a better term though; I've considered `elab_origin_span`, `while_elaborating_span`, `ast_node_span`.

Reply:
hmm, I will probably take one of your options, I like `while_elaborating_span` best, for now :)